rustls/tests/api.rs

```diff
@@ -2964,6 +2964,69 @@ fn sni_resolver_rejects_bad_certs() {
     );
 }
 
+#[test]
+#[cfg(test)]
```

Comment:
Is the `#[cfg(test)]` gate here, or on L2975 needed? It seems to me that they might be superfluous :thinking:

Reply:
Removed these in [Add tests for CertifiedKey::keys_match](https://github.com/rustls/rustls/pull/1954/commits/c5590aea7b08c003e6acad3ed9f4c1bdd8675951). Also, I added a Cargo.lock update to [4446c9f](https://github.com/rustls/rustls/pull/1954/commits/4446c9f779e5ad1053b25b486204c19940af758e)

Reply:
> Also, I added a Cargo.lock update to [4446c9f](https://github.com/rustls/rustls/pull/1954/commits/4446c9f779e5ad1053b25b486204c19940af758e)

The `fuzz` crate needs its `Cargo.lock` bumped too. I forget this roughly 50% of the time I do this dance :laughing: